File: Maratona/Embaralhando.c
#include <stdio.h>

long long fatorial(long long x, long long resposta){
    if(x != 1){
        resposta = resposta * x;
        return fatorial(x-1, resposta);
    }
    else return resposta;
}
int main(void){

    char letra;
    int checagem = scanf("%c", &letra);
    while(letra!='0' && letra != EOF && checagem > 0){
        int tamanho = 0;
        while(letra!='\n'){
            if(letra!=' '){
                tamanho++;
            }
            scanf("%c", &letra);
        }
        long long resposta = fatorial(tamanho, 1);
        printf("%llu\n", resposta);
        scanf("%c", &letra);
    }
    return 0;
}

File: Outros/Maratona/Embaralhando.py
def fatorial(x, resposta = 1):
    if x != 1:
        resposta = resposta*x
        return fatorial(x-1, resposta)
    return resposta

nTeste = input()
while nTeste != '0':
    tamanho = len(nTeste)
    answer = fatorial(tamanho)
    print(answer)
    try:
        nTeste = input()
    except EOFError:
        nTeste = '0'
